src/addons/matchrecursive.js

```diff
@@ -179,10 +180,19 @@ export default (XRegExp) => {
                         break;
                     }
                 }
-            } else {
+            } else if (unbalancedDelimiters === 'error') {
                 const delimSide = rightMatch ? 'right' : 'left';
                 const errorPos = rightMatch ? delimStart : outerStart;
                 throw new Error(`Unbalanced ${delimSide} delimiter found in string at position ${errorPos}`);
+            } else if (unbalancedDelimiters === 'text') {
+                if (rightMatch) {
+                    rightMatch = null;
```

Comment:
this is the case when it's an unbalanced right delimiter. Simply setting rightMatch to null will treat the character as raw text since all of the requisite indices were updated